engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -1240,128 +1234,182 @@ public static Clock currentClock() {
-     * {@link #currentDateNyOverride} value. If {@link #currentDateNyOverride} has not been set, this method has no
-     * effect.
+     * Converts nanos of epoch to a {@link ZonedDateTime}.
+     *
+     * @param nanos nanoseconds since epoch
+     * @param timeZone the {@link TimeZone time zone}
+     *
+     * @return a new {@link ZonedDateTime} or null if nanos was {@link QueryConstants#NULL_LONG}.
      */
-    public static void overrideLastBusinessDateNyFromCurrentDateNy() {
-        if (currentDateNyOverride != null) {
-            final BusinessCalendar bc = Calendars.calendar("USNYSE");
-            lastBusinessDayNyOverride = bc.previousBusinessDay(currentDateNyOverride.substring(0, 10));
-        }
+    @Nullable
+    public static ZonedDateTime makeZonedDateTime(final long nanos, @NotNull final TimeZone timeZone) {
+        return makeZonedDateTime(nanos, timeZone.getZoneId());
     }
 
     /**
-     * Cached value of lastBusinessDateNy, which expires after milliseconds from Epoch value of
-     * {@link #endOfCurrentDateNyLastBusinessDay}
+     * Converts nanos of epoch to a {@link ZonedDateTime}.
+     *
+     * @param nanos nanoseconds since epoch
+     * @param zone the {@link ZoneId time zone}
+     *
+     * @return a new {@link ZonedDateTime} or null if nanos was {@link QueryConstants#NULL_LONG}.
      */
-    public static String lastBusinessDateNy = null;
+    @Nullable
+    public static ZonedDateTime makeZonedDateTime(final long nanos, ZoneId zone) {
+        // noinspection ConstantConditions
+        return nanos == NULL_LONG ? null : ZonedDateTime.ofInstant(makeInstant(nanos), zone);
+    }
+
     /**
-     * Expiration for cached {@link #lastBusinessDateNy} as milliseconds from Epoch.
+     * Converts a {@link DateTime} to a {@link ZonedDateTime}.
+     *
+     * @param dateTime The a {@link DateTime} to convert.
+     * @return A {@link ZonedDateTime} using the default time zone for the session as indicated by
+     *         {@link TimeZone#TZ_DEFAULT}.
      */
-    public static long endOfCurrentDateNyLastBusinessDay = 0;
+    @Nullable
+    public static ZonedDateTime getZonedDateTime(final @Nullable DateTime dateTime) {
+        return getZonedDateTime(dateTime, TimeZone.TZ_DEFAULT);
+    }
 
     /**
-     * Provides a String representing the previous business date in the New York time zone using the NYSE calendar, or,
-     * if a custom {@link #lastBusinessDayNyOverride} has been set, the date provided by that override.
+     * Converts a {@link DateTime} to a {@link ZonedDateTime}.
      *
-     * @return A String in yyyy-MM-dd format.
+     * @param dateTime The a {@link DateTime} to convert.
+     * @param timeZone The {@link TimeZone} to use for the conversion.
+     * @return A {@link ZonedDateTime} using the specified time zone. or null if dateTime was null
      */
-    public static String lastBusinessDateNy() {
```

Comment:
Removing lastBusinessDateNy() is probably defensible if you do not want to own calendars at all (but then Enterprise needs to reimplement something).  You'll also lose a lot of plot utility w/o business calendars.

I think not having currentDateNy() which does not rely on calendars is pretty bad.  @rbasralian do you think this is a tenable position with respect to usability?

None of the override stuff I particular see us needing.

Reply:
As discussed offline, for the foreseeable future, DHE will own supporting clients in this regard.